### PR TITLE
Grab control rates with LTFU from repository

### PR DESCRIPTION
**Pull request by rsanheim**
The code for all this was mostly already here, just needs to be wired up and verified.

**Story card:** [ch3668](https://app.clubhouse.io/simpledotorg/story/3668/grab-all-ltfu-data-from-repository-instead-of-regionservice)

### CODE REVIEW

**Review by timcheadle — APPROVED** [review]

File: app/models/reports/repository.rb [hunk 1]
```diff
@@ -233,7 +233,7 @@ def initialize(regions, periods:)
     end
 
     memoize def controlled_rates(with_ltfu: false)
-      region_period_cached_query(__method__, with_ltfu: false) do |entry|
+      region_period_cached_query(__method__, with_ltfu: with_ltfu) do |entry|
```
Comment (harimohanraj89) [inline app/models/reports/repository.rb:236]: 💯

File: app/services/control_rate_service.rb [hunk 2]
```diff
@@ -61,10 +61,9 @@ def fetch_all_data
 
     results.controlled_patients_rate = repository.controlled_rates[slug]
     results.uncontrolled_patients_rate = repository.uncontrolled_rates[slug]
-
-    results.calculate_percentages(:controlled_patients, with_ltfu: true)
-    results.calculate_percentages(:uncontrolled_patients, with_ltfu: true)
-    results.calculate_percentages(:ltfu_patients)
+    results.controlled_patients_with_ltfu_rate = repository.controlled_rates(with_ltfu: true)[slug]
+    results.uncontrolled_patients_with_ltfu_rate = repository.uncontrolled_rates(with_ltfu: true)[slug]
+    results.ltfu_patients_rate = repository.ltfu_rates[slug]
```
Comment (harimohanraj89) [inline app/services/control_rate_service.rb:66]: This is just switching to the more explicit way of setting these repo keys? Consistent with above?

**Review by harimohanraj89 — APPROVED** [review]
Lookin' slick

File: app/services/control_rate_service.rb (hunk 2 above)
Comment (rsanheim) [reply app/services/control_rate_service.rb:66]: Yup, and its removing the responsibility from the Result object for calculating these rates.

**Pull request merged**